posthog/warehouse/api/saved_query.py

```diff
@@ -165,17 +203,33 @@ def update(self, instance: Any, validated_data: Any) -> Any:
         was_sync_frequency_updated = False
 
         with transaction.atomic():
+            locked_instance = DataWarehouseSavedQuery.objects.select_for_update().get(pk=instance.pk)
```

Comment:
If something in here raises an exception, is the lock released?

Reply:
yeah, it'll release when the transaction finishes which means it commits or rollsback. (Pending question for @tomasfarias about the one try/catch)